lib/web/assistant.go

```diff
@@ -19,46 +19,785 @@
 package web
 
 import (
+	"context"
+	"encoding/json"
+	"io"
 	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"time"
 
+	"github.com/gorilla/websocket"
 	"github.com/gravitational/trace"
+	"github.com/gravitational/trace/trail"
 	"github.com/julienschmidt/httprouter"
+	"github.com/sashabaranov/go-openai"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/gen/proto/go/assist/v1"
+	pluginsv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	usageeventsv1 "github.com/gravitational/teleport/api/gen/proto/go/usageevents/v1"
+	"github.com/gravitational/teleport/lib/ai"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/httplib"
 	"github.com/gravitational/teleport/lib/reversetunnel"
 )
 
+// assistantMessageType is a type of the Assist message.
+type assistantMessageType string
+
+const (
+	// messageKindCommand is the type of Assist message that contains the command to execute.
+	messageKindCommand assistantMessageType = "COMMAND"
+	// messageKindCommandResult is the type of Assist message that contains the command execution result.
+	messageKindCommandResult assistantMessageType = "COMMAND_RESULT"
+	// messageKindUserMessage is the type of Assist message that contains the user message.
+	messageKindUserMessage assistantMessageType = "CHAT_MESSAGE_USER"
+	// messageKindAssistantMessage is the type of Assist message that contains the assistant message.
+	messageKindAssistantMessage assistantMessageType = "CHAT_MESSAGE_ASSISTANT"
+	// messageKindAssistantPartialMessage is the type of Assist message that contains the assistant partial message.
+	messageKindAssistantPartialMessage assistantMessageType = "CHAT_PARTIAL_MESSAGE_ASSISTANT"
+	// messageKindAssistantPartialFinalize is the type of Assist message that ends the partial message stream.
+	messageKindAssistantPartialFinalize assistantMessageType = "CHAT_PARTIAL_MESSAGE_ASSISTANT_FINALIZE"
+	// messageKindSystemMessage is the type of Assist message that contains the system message.
+	messageKindSystemMessage assistantMessageType = "CHAT_MESSAGE_SYSTEM"
+)
```

Comment:
Aren't these types going to be useful outside of web package e.g. where assistant types/messages are defined so backend services can use them too? Or you don't anticipate them being needed anywhere else?